hcl2template/types.packer_config.go

```diff
@@ -622,6 +622,7 @@ func (cfg *PackerConfig) GetBuilds(opts packer.GetBuildsOptions) ([]packersdk.Bu
 
 			decoded, _ := decodeHCL2Spec(srcUsage.Body, cfg.EvalContext(BuildContext, nil), builder)
 			pcb.HCLConfig = decoded
+			pcb.BuilderType = srcUsage.Type
```

Comment:
Is this field really necessary? 

Why not have a method on SourceRef that returns the underlying component type string?